examples/bpi-monolith-test-suite/bpi/components/zkmock/agreementalt.ts

```diff
@@ -0,0 +1,130 @@
+import { OrderAlt } from "./orderalt";
+import { MerkleTree } from 'merkletreejs'
+import sha256 = require('crypto-js/sha256');
+import { Shield } from "./shield";
+import { Verifier } from "./verifier";
+import { placeOrderPredicate, placeOrderInput } from "./predicates";
+
+export class AgreementAlt {
+
+    productIds: string[];
+
+    order: OrderAlt;
+
+    proofs: string[] = [];
+
+    buyerPK: string; // In practice: curve point
+
+    sellerPK: string; // As above
+
+    shield: Shield;
+
+    verifiers: [];
+
+    // Sets initial aggreement state (counterparty  public keys, product ids) and deploys shield + verifier contracts
+    constructor(sellerPK: string, buyerPK: string, productIds: string[]) {
+        this.productIds = productIds;
+        this.buyerPK = buyerPK;
+        this.sellerPK = sellerPK;
+    }
+
+    public deployShield(): void {
```

Comment:
soooo ... there is no concept of a shield in the standard. Is this supposed to be a security policy? again abstract naming and remove implementation artifacts as much as possible. @ognjenkurtic